src/frame/endpoints/UserApiEndpoints.ts

```diff
@@ -307,7 +318,28 @@ export default {
         if (!userList.length) {
             return Future.of({result: []});
         }
-        const {url, options, errorCode} = userKeyList(userList);
-        return makeAuthorizedApiRequest(url, errorCode, options);
+
+        // Check the list for any users we already have the public key cached for
+        const cacheHits: UserKeyListResponseObject[] = [];
+        userList.forEach((userId) => {
+            const userMasterPublicKey = userPublicKeyCache[userId];
+            if (userMasterPublicKey) {
+                cacheHits.push(userMasterPublicKey);
+            }
+        });
+
+        // If we have any userIds that weren't cached, fetch all of them
+        if (cacheHits.length === userList.length) {
+            return Future.of({result: cacheHits});
```

Comment:
Yeah. Here we're replicating the original API response, just swapping it with the cached values, since we're doing it inside the original API call (because it was possible to). In the group situation it had to be in a new call that is called in spots that only care about the public key, so it could easily be a new, more tailored response.